opm/simulators/flow/KeywordValidation.cpp

```diff
@@ -75,13 +92,23 @@ namespace KeywordValidation
     }
 
     void
-    KeywordValidator::validateDeck(const Deck& deck, const ParseContext& parse_context, ErrorGuard& error_guard) const
+    KeywordValidator::validateDeck(const Deck& deck,
+                                   const std::unordered_map<std::string, std::function<void(const DeckKeyword& keyword, std::vector<ValidationError>& errors)>>& special_validation,
```

Comment:
We could also pass the map of functions in the constructor of the validation object. That is also where the other validation info is passed, so it may be a bit more logical. But passing it here works also fine, of course. What do you think?

Reply:
That is a good idea - Will upfate.